test/jest/unit/handlers/quote.test.ts

```diff
@@ -0,0 +1,156 @@
+    });
+
+    it('works with other chains', () => {
+      expect(
+        QuoteHandler.protocolsFromRequest(
+          ChainId.BASE,
+          ['v2', 'v3', 'mixed'],
+          '',
+          undefined,
+          undefined
+        )
+      ).toEqual([Protocol.V2, Protocol.V3, Protocol.MIXED]);
+    });
+
+    it('returns empty when a requested protocol is invalid', () => {
+      expect(
+        QuoteHandler.protocolsFromRequest(
+          ChainId.BASE,
+          ['v2', 'v3', 'mixed', 'miguel'],
+          '',
+          undefined,
+          undefined
+        )
+      ).toEqual([]);
+    });
+
+    describe('for mobile request', () => {
+      it('removes v2 and mixed with other chains, when the requested source is mobile', () => {
+        ['uniswap-ios', 'uniswap-android'].forEach(requestSource => {
+          expect(
+            QuoteHandler.protocolsFromRequest(
+              ChainId.BASE,
+              ['v2', 'v3', 'mixed'],
+              requestSource,
+              undefined,
+              undefined
+            )
+          ).toEqual([Protocol.V3]);
+        });
+      });
+
+      it('removes v2 and mixed with other chains, when the requested source is mobile, and version 1.23, 1.23.5 or 1.24', () => {
+        ['uniswap-ios', 'uniswap-android'].forEach(requestSource => {
+          ['1.23', '1.23.5', '1.24'].forEach(appVersion => {
+            expect(
+              QuoteHandler.protocolsFromRequest(
+                ChainId.BASE,
+                ['v2', 'v3', 'mixed'],
+                requestSource,
+                appVersion,
+                undefined
+              )
+            ).toEqual([Protocol.V3]);
+          });
+        });
+      });
+
+      it('allows v2 and mixed with mainnet, even when the requested source is mobile, and version 1.23, 1.23.5 or 1.24', () => {
+        ['uniswap-ios', 'uniswap-android'].forEach(requestSource => {
+          ['1.23', '1.23.5', '1.24', '1.24', '1.24.test'].forEach(appVersion => {
+            expect(
+              QuoteHandler.protocolsFromRequest(
+                ChainId.MAINNET,
+                ['v2', 'v3', 'mixed'],
+                requestSource,
+                appVersion,
+                undefined
+              )
+            ).toEqual([Protocol.V2, Protocol.V3, Protocol.MIXED]);
+          });
+        });
+      });
+
+      it('allows v2 and mixed with other chains, when the requested source is mobile, and version is 1.25 or greater', () => {
+        ['uniswap-ios', 'uniswap-android'].forEach(requestSource => {
+          ['1.25', '1.25.5', '1.26', '1.26.test'].forEach(appVersion => {
+            expect(
+              QuoteHandler.protocolsFromRequest(
+                ChainId.BASE,
+                ['v2', 'v3', 'mixed'],
+                requestSource,
+                appVersion,
+                undefined
+              )
+            ).toEqual([Protocol.V2, Protocol.V3, Protocol.MIXED]);
+          });
+        });
+      });
+    });
+  });
```

Comment:
@adjkant are these the kind of versions that I can expect being sent in the x-app-version header from the mobile app?

Reply:
Yes except for .test!

Reply:
would you ever have something like `.build1`?